### ISSUE TITLE
val loss is low but visiualize is bad?

### ISSUE THREAD

**Issue by jinfagang** [zylo117/Yet-Another-EfficientDet-Pytorch#295]
I got some val loss like :

```
checkpoint...
Step: 46255. Epoch: 6/500. Iteration: 6608/6608. Cls loss: 0.18054. Reg loss: 0.02094. Total loss: 0.20148: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6608/6608 [2:08:43<00:00,  1.17s/it]
Val. Epoch: 6/500. Classification loss: 0.03633. Regression loss: 0.00822. Total loss: 0.04455
Step: 46499. Epoch: 7/500. Iteration: 244/6608. Cls loss: 0.00766. Reg loss: 0.00198. Total loss: 0.00964:   4%|█████▎                                                                                                                                           | 243/6608 [04:48<2:11:44,  1.24s/it]checkpoint...
Step: 46999. Epoch: 7/500. Iteration: 744/6608. Cls loss: 0.00368. Reg loss: 0.00297. Total loss: 0.00665:  11%|████████████████▎                                                                                                                                | 743/6608 [14:41<1:54:09,  1.17s/it]checkpoint...
Step: 47086. Epoch: 7/500. Iteration: 831/6608. Cls loss: 0.03998. Reg loss: 0.00191. Total loss: 0.04190:  13%|██████████████████▏                                                                                                                              | 831/6608 [16:23<1:47:45,  1.12s/it]Step: 47087. Epoch: 7/500. Iteration: 832/6608. Cls loss: 0.03041. Reg loss: 0.01038. Total loss: 0.04079:  13%|██████████████████▎                                                                                                                              | 832/6608 [16:24<1:46:59,  1.11s/it]Step: 47499. Epoch: 7/500. Iteration: 1244/6608. Cls loss: 0.00419. Reg loss: 0.00390. Total loss: 0.00809:  19%|
```

Does the val loss is low enough? But my visiualize result is not accurate at all.

**Comment by shengyuqing** [comment]
same problem.

**Comment by jinfagang** [comment]
Oh, it related with many mistakes.

Anyone got this issue probably not change anchor as training. Also be carefully coco using category_id from 1, while most your dataset may from 0, so -1 is not nesscery.
